parsers/src/main/scala/uk/gov/ons/addressIndex/crfscala/CrfParser.scala

```diff
@@ -9,12 +9,23 @@ trait CrfParser {
   val crfScala: CrfScalaJniImpl = new CrfScalaJniImpl()
 
   //TODO scaladoc
-  def tag(i: Input, features: CrfFeatures, tokenable: CrfTokenable): String = {
+  def tag(input: Input, features: CrfFeatures, tokenable: CrfTokenable): Seq[CrfTokenResult] = {
+    //todo optimise file
     val currentDirectory = new java.io.File(".").getCanonicalPath
     val modelPath = s"$currentDirectory/parsers/src/main/resources/addressCRFA.crfsuite"
-    val actual = parse(i, features, tokenable)
+    val actual = parse(input, features, tokenable)
     val augmentedActual = augmentCrfJniInput(actual)
```

Comment:
https://www.google.co.uk/search?sourceid=chrome-psyapi2&ion=1&espv=2&ie=UTF-8&q=augmented%20definition&oq=augmented%20definition&aqs=chrome..69i57j0l5.2231j0j7 

the context is being provided by `augmentCrfJniInput`